modules/claims_api/spec/lib/claims_api/v2/disability_compensation_validation_spec.rb

```diff
@@ -65,40 +65,47 @@ class TestDisabilityCompensationValidationClass
       ] }
     end
 
-    let(:service_period_end_in_future) do
-      { 'servicePeriods' => [
-        {
-          'serviceBranch' => 'Public Health Service',
-          'serviceComponent' => 'Active',
-          'activeDutyBeginDate' => '2008-11-14',
-          'activeDutyEndDate' => 2.days.from_now.strftime('%Y-%m-%d').to_s
-        }
-      ] }
-    end
-
     let(:service_periods) { form_attributes['serviceInformation'] }
 
     # rubocop:disable RSpec/SubjectStub
-    context 'calls to retrieve codes' do
+    context 'when a separation location code is present' do
       before do
-        allow(test_526_validation_instance).to receive(:retrieve_separation_locations).and_return([])
+        separation_locations = [
+          { id: 24_912, description: 'AF Academy' },
+          { id: 26_722, description: 'ANG Hub' }
+        ]
+        allow(test_526_validation_instance).to receive(:retrieve_separation_locations)
+          .and_return(separation_locations)
       end
 
-      it 'retrives codes if separationLocationcode is present' do
-        test_526_validation_instance.send(:validate_form_526_location_codes, service_periods)
-        expect(test_526_validation_instance).to have_received(:retrieve_separation_locations)
+      context 'when the location code is valid' do
+        it 'returns no errors' do
+          service_periods['servicePeriods'][0]['separationLocationCode'] = '24912'
+          test_526_validation_instance.send(:validate_form_526_location_codes, service_periods)
+          errors = test_526_validation_instance.send(:error_collection)
+
+          expect(errors).to be_empty
+        end
+      end
+
+      context 'when the location code is invalid' do
+        it 'adds an error to the errors array' do
+          service_periods['servicePeriods'][0]['separationLocationCode'] = '123456'
+          test_526_validation_instance.send(:validate_form_526_location_codes, service_periods)
+          errors = test_526_validation_instance.send(:error_collection)
+
+          expect(errors.size).to eq(1)
+        end
```

Comment:
_Might_ be worth it to add a test handling multiple service periods with mixed results as well (one good one errored, shows correct index, that kind of thing) <---  _Really more of a nit though_

Reply:
https://github.com/department-of-veterans-affairs/vets-api/pull/18258/commits/9f86d19bb783240221bd786d11eb2ff21f037de1